### PR TITLE
CANTINA AUDIT #7 - RLP Changes

### PR DESCRIPTION
**Pull request by johnwhitton**
Apply changes recommended by Cantina and held in this [Pull Request](https://github.com/eco/eco-routes/pull/138)

The pull request uses a more consistent approach to a couple of edge cases around leading zeroes.

This was modified and tested under the [AUDIT-RLP-TESTS](https://github.com/eco/eco-routes/tree/AUDIT-RLP-TESTS) branch.

### CODE REVIEW

File: contracts/Prover.sol [hunk 1]
```diff
@@ -342,26 +357,24 @@ contract Prover is SimpleProver {
...
-                        createdAt
+    ) public pure returns (bytes32 gameStatusStorageSlotRLP) {
+      // Packed data is 64 + 64 + 8 + 8 + 8 = 152 bits / 19 bytes.
+      // Need to convert to `uint152` to preserve right alignment.
+        return bytes32(
+            uint256(
+                uint152(
+                    bytes19(
+                        abi.encodePacked(
+                            l2BlockNumberChallenged,
```
Comment (nknavkal) [inline contracts/Prover.sol:368]: does this account for both of the cases you had before?

**Review by nknavkal — APPROVED** [review]
looks good i think

File: contracts/Prover.sol (hunk 1 above)
Comment (johnwhitton) [reply contracts/Prover.sol:368]: Yes tested it with leading zero in the [AUDIT-RLP-TESTS](https://github.com/eco/eco-routes/tree/AUDIT-RLP-TESTS) branch SEE [here]([AUDIT-RLP-TESTS](https://github.com/eco/eco-routes/tree/AUDIT-RLP-TESTS)) and [here](https://github.com/eco/eco-routes/blob/AUDIT-RLP-TESTS/test/StorageRLP.spec.ts#L312-L415)
